app/scripts/components/exploration/hooks/use-analysis-data-request.ts

```diff
@@ -105,25 +106,44 @@ export function useAnalysisDataRequest({
       type: 'FeatureCollection',
       features: selectedFeatures
     };
-
     const { start, end } = selectedInterval;
-
-    requestDatasetTimeseriesData({
-      maxItems: MAX_QUERY_NUM,
-      start,
-      end,
-      aoi,
-      dataset,
-      queryClient,
-      concurrencyManager: analysisConcurrencyManager,
-      onProgress: (data) => {
-        setAnalysis(data);
-      }
-    });
+      (async () => {
+        await requestDatasetTimeseriesData({
+        maxItems: MAX_QUERY_NUM,
+        start,
+        end,
+        aoi,
+        dataset,
+        queryClient,
+        concurrencyManager: analysisConcurrencyManager,
+        onProgress: (data) => {
+          queueMicrotask(() => {
+            flushSync(() => {
+              setAnalysis(data);
+            });
+          });
+        }
+      });
+      // setAnalysis(stat);
+    })();
     // We want great control when this effect run which is done by incrementing
     // the analysisRun. This is done when the user refreshes the analysis or
     // when they enter the analysis. It is certain that when this effect runs
     // the other values will be up to date. Adding all dependencies would cause
     // the hook to continuously run.
   }, [analysisRunId, datasetStatus, isAnalyzing]);
+
+
+  // useEffect(() => {
+  //   // TECH DEBT
+  //   // setAnalysis sets Jotai Atom value for analysis result
+  //   // It should sequentially set the analysis progress as 
+  //   // idle => loading (increasing number of loaded items) => success
+  //   // However, it is not setting the value sequentially, 
+  //   // ended up loading status as a result even when the request is successful
+  //   // This just overwrites the analysis result with the final result of analysis, 
+  //   // so Jotai value gets 'corrected' in case it is wrong
+  //   setAnalysis(analysisResult);
+  // },[setAnalysis, analysisResult]);
```

Comment:
I would leave an `@NOTE` OR `@TECH-DEBT` comment to explain why we have to brute force overwrite